cloud/deploy/bundle.go

```diff
@@ -0,0 +1,247 @@
+	if err != nil {
+		return err
+	}
+
+	// finalize the deploy
+	_, err = finalizeBundleDeploy(c.Organization, deploy.Id, tarballVersion, input, coreClient)
+	if err != nil {
+		return err
+	}
+	fmt.Println("Successfully uploaded bundle with version " + tarballVersion + " to Astro.")
+
+	// if requested, wait for the deploy to finish by polling the deployment until it is healthy
+	if input.Wait {
+		err = deployment.HealthPoll(currentDeployment.Id, currentDeployment.WorkspaceId, dagOnlyDeploySleepTime, tickNum, timeoutNum, platformCoreClient)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func uploadBundle(tarDirPath, bundlePath, uploadURL string, prependBaseDir bool) (string, error) {
+	tarFilePath := filepath.Join(tarDirPath, "bundle.tar")
+	defer func() {
+		err := os.Remove(tarFilePath)
+		if err != nil {
+			fmt.Println("\nFailed to delete tar file: ", err.Error())
+			fmt.Println("\nPlease delete the tar file manually from path: " + tarFilePath)
+		}
+	}()
+
+	// Generate the bundle tar
+	err := fileutil.Tar(bundlePath, tarFilePath, prependBaseDir)
+	if err != nil {
+		return "", err
+	}
+
+	tarFile, err := os.Open(tarFilePath)
+	if err != nil {
+		return "", err
+	}
+	defer tarFile.Close()
+
+	versionID, err := azureUploader(uploadURL, tarFile)
+	if err != nil {
+		return "", err
+	}
+
+	return versionID, nil
+}
+
+func getDeployment(organizationID, deploymentID string, platformCoreClient astroplatformcore.CoreClient) (*astroplatformcore.Deployment, error) {
+	resp, err := platformCoreClient.GetDeploymentWithResponse(context.Background(), organizationID, deploymentID)
+	if err != nil {
+		return nil, err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	return resp.JSON200, nil
+}
+
+func createBundleDeploy(organizationID string, input *DeployBundleInput, deployGit *astrocore.DeployGit, coreClient astrocore.CoreClient) (*astrocore.Deploy, error) {
+	request := astrocore.CreateDeployRequest{
+		Description:     &input.Description,
+		Type:            astrocore.CreateDeployRequestTypeBUNDLE,
+		BundleMountPath: &input.MountPath,
+		BundleType:      &input.BundleType,
+	}
+	if deployGit != nil {
+		request.Git = &astrocore.CreateDeployGitRequest{
+			Provider:   astrocore.CreateDeployGitRequestProvider(deployGit.Provider),
+			Repo:       deployGit.Repo,
+			Account:    deployGit.Account,
+			Path:       deployGit.Path,
+			Branch:     deployGit.Branch,
+			CommitSha:  deployGit.CommitSha,
+			CommitUrl:  fmt.Sprintf("https://github.com/%s/%s/commit/%s", deployGit.Account, deployGit.Repo, deployGit.CommitSha),
+			AuthorName: deployGit.AuthorName,
+		}
+	}
+	resp, err := coreClient.CreateDeployWithResponse(context.Background(), organizationID, input.DeploymentID, request)
+	if err != nil {
+		return nil, err
+	}
+	err = astrocore.NormalizeAPIError(resp.HTTPResponse, resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	return resp.JSON200, nil
+}
```

Comment:
Similar here, rather than a whole new function, could this be merged in to `createDeploy` in this package (inside `./deploy.go`)?